### ISSUE TITLE
Allow eslint checker should read .eslintignore

### ISSUE THREAD

**Issue by wyuenho** [flycheck/flycheck#1193]
The eslint checker command currently only looks for the .eslintrc file by traversing up the directory hierarchy of the buffer file path but it doesn't seem to use the .eslintignore file when found in a directory. So even when eslint offers the ability to disable checking for some files, the flycheck eslint checker doesn't seem able to.

**Comment by cpitclaudel** [comment]
Thanks for this report.  Can you clarify what's going wrong, what you were expecting, and what happens instead? Ideally with an example (I'm not very familiar with eslint).

**Comment by Simplify** [comment]
@wyuenho Flycheck doesn't search for `.eslintrc` or `.eslintignore`, we just run `eslint` command with filename as argument and ESLint does the rest, like a finding his configuration file.

[ESLint ignoring files and directories](http://eslint.org/docs/user-guide/configuring#ignoring-files-and-directories)

This is "problem" with ESLint. When I place `.eslintignore` in project root directory:
```
$ eslint src/js/simplify.js --format=checkstyle
<?xml version="1.0" encoding="utf-8"?><checkstyle version="4.3"><file name="[path-to-project-root]/src/js/simplify.js"><error line="undefined" column="undefined" severity="warning" message="File ignored because of a matching ignore pattern. Use &quot;--no-ignore&quot; to override." source="" /></file></checkstyle>
```

When I place `.eslintignore` is <project-root>/src/:
```
$ eslint src/js/simplify.js --format=checkstyle
<?xml version="1.0" encoding="utf-8"?><checkstyle version="4.3"><file name="[path-to-project-root]/src/js/simplify.js"><error line="13" column="5" severity="error" message="Parsing error: Unexpected token, expected ," source="" /></file></checkstyle>
```

We can add `--ignore-path` option to `eslint` checker, but that doesn't solve your problem if you have multiple directories with `.eslintignore` file.

Solution: Create single `.eslintignore` file in same directory where your ESLint config is located. Add all paths from sub-directories in there. For example:
```
build/*
test/*
vendor/*
```

My guess is that ESLint search for `.eslintignore` file in the same place where `.eslintrc` is located.

**Comment by wyuenho** [comment]
I do have a single .eslintignore file on the top of the working directory. I'll circle back with a test case.

**Comment by wyuenho** [comment]
You are right, this seems to be a problem with eslint. I'll file a bug there.

**Comment by ggb667** [comment]
This is also broken for me.  Anyone have a work-around?
